ros2doctor/ros2doctor/api/platform.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import platform
+import sys
+
+import rosdistro
+
+
+def print_platform_info():
+    """Print out platform information."""
+    platform_name = platform.system()
+    # platform info
+    print('PLATFORM INFORMATION')
+    print('system               : ', platform_name)
+    print('Platform Info        : ', platform.platform())
+    if platform_name == 'Darwin':
+        print('Mac OS version       : ', platform.mac_ver())
+    print('release              : ', platform.release())
+    print('processor            : ', platform.processor())
+
+    # python info
+    print('PYTHON INFORMATION')
+    print('version              : ', platform.python_version())
+    print('compiler             : ', platform.python_compiler())
+    print('build                : ', platform.python_build())
```

Comment:
The `ros2 doctor --report` output looks great!

```
$ ros2 doctor --report
PLATFORM INFORMATION
system               :  Linux
Platform Info        :  Linux-4.15.0-55-generic-x86_64-with-Ubuntu-18.04-bionic
release              :  4.15.0-55-generic
processor            :  x86_64
PYTHON INFORMATION
version              :  3.6.8
compiler             :  GCC 8.0.1 20180414 (experimental) [trunk revision 259383
build                :  ('default', 'Jan 14 2019 11:02:34')
ROS INFORMATION
distribution name    :  dashing
distribution type    :  ros2
distribution status  :  active
release platforms    :  {'ubuntu': ['bionic']}
```

People that add report entry points will want to match the indentation to keep a consistent look. They'll have to make their reports pad spaces out to 21 characters before a `:` and then two spaces after the `:` before the content. If we want to make that longer in the future it might be a minor burden to update the spacing everywhere.

It might be a good idea to separate the concern of presenting data from getting the data. If the report entry points returned their data in a common format, then `generate_report()` could be made responsible for figuring out how to display it.

Design-wise I think it's fine as it is for now. This is just something to think about as an enhancement.

Reply:
Good point! Added to my features list.